pkg/reconciler/ingress/ingress.go

```diff
@@ -196,6 +202,27 @@ func gatherProbes(b *status.Backends, r *gatewayapi.HTTPRoute, visibility v1alph
 	}
 }
 
+// determineLoadBalancerIngressStatus will return the address for the Gateway.
+// If a service is provided, it will return the address of the service.
+// Otherwise, it will return the first address in the Gateway status.
+func (c *Reconciler) determineLoadBalancerIngressStatus(gwc config.GatewayConfig) ([]v1alpha1.LoadBalancerIngressStatus, error) {
+	if gwc.Service != nil {
+		return []v1alpha1.LoadBalancerIngressStatus{
+			{DomainInternal: network.GetServiceHostname(gwc.Service.Name, gwc.Service.Namespace)},
+		}, nil
+	}
+	gw, err := c.gatewayLister.Gateways(gwc.Gateway.Namespace).Get(gwc.Gateway.Name)
+	if apierrs.IsNotFound(err) {
+		return nil, fmt.Errorf("Gateway %s does not exist: %w", gwc.Gateway.Name, err) //nolint:stylecheck
+	} else if err != nil {
+		return nil, err
+	}
+
+	return []v1alpha1.LoadBalancerIngressStatus{
+		{DomainInternal: gw.Status.Addresses[0].Value},
```

Comment:
```suggestion
		{IP: gw.Status.Addresses[0].Value},
```
@dprotaso are there some cases where the gateway returns a domain instead of IP?

As I recall, on AWS... creating a loadbalancer type service would have resulted in a domain rather than an IP.

Reply:
Yeah we should inspect the address type here

https://github.com/kubernetes-sigs/gateway-api/blob/90e22fe09c18b3d35aa79917ed50dd2a3e055666/apis/v1/gateway_types.go#L615

Reply:
hmmm yeahh good point. I can just make it switch on Addresses[0] type